flocker/control/test/test_httpapi.py

```diff
@@ -2628,22 +2629,24 @@ def test_returns_service(self):
         """
         reactor = MemoryReactor()
         endpoint = TCP4ServerEndpoint(reactor, 6789)
-        verifyObject(IService, create_api_service(None, None, endpoint))
+        verifyObject(IService, create_api_service(
+            None, None, endpoint, ClientContextFactory()))
```

Comment:
This call violates the documented interface (`None` is not allowed for those parameters).